compiler/src/language_server/sourcetree.re

```diff
@@ -428,7 +442,10 @@ module Sourcetree: Sourcetree = {
             | Some(path) =>
               let decl = Env.find_type(path, pat.pat_env);
               if (decl.type_loc == Location.dummy_loc) {
-                None;
+                switch (pat.pat_desc) {
+                | TPatConstruct(_, desc, _) => Some(desc.cstr_loc)
+                | _ => None
+                };
```

Comment:
Why is this only done when the `type_loc` is a dummy_loc?

Reply:
I think it was just a backup, for if it had a dummy loc because I was running into issues.

Reply:
That doesn't help. I want to know why there was a dummy_loc... what causes this? does a larger fix need to be investigated? Is it unrelated to the PR?

Reply:
This wasnt neccessairy I think i was just being precautious when i added that. so i removed it